File: _stats_build/statistics/wr_ao4r.rb
require_relative "abstract/ao_rounds"

class WrAo4r < AoRounds
  def initialize
    @title = "World record Ao4R (Average of 4 Rounds) history"
    @note = "Ao4R: average of the 4 round averages (R1 + R2 + R3 + Final) in a competition."
    @table_header = { "Result" => :right, "Improvement" => :right, "Days" => :right, "Person" => :left, "Date" => :left, "Competition" => :left, "Details" => :left }
  end

  def round_count
    4
  end
end

File: core/jobs/stats-build/src/statistics/wr_ao4r.ts
// NOTE: Ao4R——一场比赛恰好 4 轮时各轮 average 的均值
import { AoRounds } from '../core/ao_rounds.js';

export class WrAo4r extends AoRounds {
  constructor() {
    super();
    this.title = 'Ao4R';
    this.titleZh = 'Ao4R';
    this.note = 'Average of 4 Rounds: mean of averages when a person competed in exactly 4 rounds at a competition.';
    this.noteZh = '四轮平均：一场比赛中恰好参加 4 轮时各轮平均的均值。';
  }
  roundCount() { return 4; }
}
